dask/array/wrap.py

```diff
@@ -39,8 +40,10 @@ def wrap_func_shape_as_first_arg(func, *args, **kwargs):
 
     keys = product([name], *[range(len(bd)) for bd in chunks])
     shapes = product(*chunks)
-    func = partial(func, dtype=dtype, **kwargs)
-    vals = ((func,) + (s,) + args for s in shapes)
+    if 'dtype' not in kwargs:
+        kwargs['dtype'] = dtype
```

Comment:
I prefer

`kwargs.setdefault('dtype', dtype)`

but not everyone may be familiar with that method.